### PR TITLE
[4.0] [plg_system_httpheaders] Add filters and validation to manifest file

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes

Adds filters and validation to manifest and subform.

### Testing Instructions

Apply patch.
Save plugin configuration if this previously saved.

### Expected result

Plugin should work like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: plugins/system/httpheaders/httpheaders.php [hunk 1]
```diff
@@ -303,9 +301,8 @@ private function setCspHeader($cspNonce)
 	 */
 	private function setHstsHeader()
 	{
-		$maxAge        = (int) $this->params->get('hsts_maxage', 31536000);
 		$hstsOptions   = array();
-		$hstsOptions[] = $maxAge < 300 ? 'max-age=300' : 'max-age=' . $maxAge;
+		$hstsOptions[] = 'max-age=' . $this->params->get('hsts_maxage', 31536000);
```
Comment (zero-24) [inline plugins/system/httpheaders/httpheaders.php]: please add the cast to integer back here.

**Review by zero-24 — CHANGES_REQUESTED** [review]
as noted inline

File: plugins/system/httpheaders/httpheaders.php (hunk 1 above)
Comment (SharkyKZ) [reply plugins/system/httpheaders/httpheaders.php]: Why tho? It's filtered in plugin configuration form.

File: plugins/system/httpheaders/httpheaders.php [hunk 2]
```diff
@@ -303,9 +301,8 @@ private function setCspHeader($cspNonce)
...
 
 		if ($this->params->get('hsts_subdomains', 0))
```
Comment (Quy) [inline plugins/system/httpheaders/httpheaders.php]: Make the same change to lines 307 & 312.

Comment (SharkyKZ) [reply plugins/system/httpheaders/httpheaders.php]: Which change?

Comment (Quy) [inline plugins/system/httpheaders/httpheaders.php]:
```suggestion
		if ($this->params->get('hsts_subdomains', 0) === 1)
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7d86048913a5cda21a011c15e23a429e3b5ca67c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25689">issues.joomla.org/tracker/joomla-cms/25689</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 3696f474121aac957d01cd98e7f73fe8594956a6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25689">issues.joomla.org/tracker/joomla-cms/25689</a>.</sub>

File: plugins/system/httpheaders/httpheaders.php (hunk 1 above)
Comment (SharkyKZ) [reply plugins/system/httpheaders/httpheaders.php]: Reverted.

**Review by zero-24 — APPROVED** [review]
LGTM

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**